src/sql/src/plan/statement/show.rs

```diff
@@ -409,25 +412,48 @@ fn show_sinks<'a>(
     scx: &'a StatementContext<'a>,
     full: bool,
     from: Option<UnresolvedObjectName>,
+    in_cluster: Option<ResolvedClusterName>,
     filter: Option<ShowStatementFilter<Aug>>,
 ) -> Result<ShowSelect<'a>, anyhow::Error> {
-    let schema = if let Some(from) = from {
-        scx.resolve_schema(from)?
-    } else {
-        scx.resolve_active_schema()?
+    let mut query_filters = vec![];
+
+    if let Some(from) = from {
+        query_filters.push(format!("schema_id = {}", scx.resolve_schema(from)?.id()));
+    } else if in_cluster.is_none() {
+        query_filters.push(format!("schema_id = {}", scx.resolve_active_schema()?.id()));
     };
 
+    if let Some(cluster) = in_cluster {
+        scx.require_experimental_mode("SHOW SINKS...IN CLUSTER")?;
+        query_filters.push(format!("clusters.id = {}", cluster.0));
+    }
+
+    let query_filters = itertools::join(query_filters.iter(), " AND ");
+
     let query = if full {
         format!(
-            "SELECT name, mz_internal.mz_classify_object_id(id) AS type, volatility
-            FROM mz_catalog.mz_sinks
-            WHERE schema_id = {}",
-            schema.id(),
+            "SELECT
+            clusters.name AS in_cluster,
```

Comment:
I think this should be called just `cluster` for consistency? (We don't say `from_schema` for schemas or `for_table` for tables, for example.)

Reply:
Hm, also, my brain wants this column to go after the cluster name. Matches the order of `CREATE SINK <name> IN CLUSTER <cluster>`, if nothing else. Thoughts?

Reply:
There's a structural taxonomy in `SHOW INDEXES`: the object it's on, followed by the index name itself. I was looking to mirror that for both `SHOW SINKS` and `SHOW INDEXES`, to reinforce the relationship between the objects and clusters--vaguely the same idea for the name `in_cluster` (teaching people how to think of these things), but glad to amend that to simply be `cluster.`

Reply:
👍🏽 no, happy to defer to your more structured reasoning!